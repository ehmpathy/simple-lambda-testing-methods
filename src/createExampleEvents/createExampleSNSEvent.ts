import { SNSEvent } from 'aws-lambda';
import { EVENT_METADATA_WHICH_SHOULD_NOT_BE_NEEDED } from './eventMetadataWhichShouldNotBeNeeded';

export const createExampleSNSEvent = ({ messages }: { messages: string[] }): SNSEvent => {
  return {
    Records: messages.map((message) => {
      return {
        EventVersion: EVENT_METADATA_WHICH_SHOULD_NOT_BE_NEEDED,
        EventSubscriptionArn: EVENT_METADATA_WHICH_SHOULD_NOT_BE_NEEDED,
        EventSource: EVENT_METADATA_WHICH_SHOULD_NOT_BE_NEEDED,
        Sns: {
          SignatureVersion: EVENT_METADATA_WHICH_SHOULD_NOT_BE_NEEDED,
          Timestamp: EVENT_METADATA_WHICH_SHOULD_NOT_BE_NEEDED,
          Signature: EVENT_METADATA_WHICH_SHOULD_NOT_BE_NEEDED,
          SigningCertUrl: EVENT_METADATA_WHICH_SHOULD_NOT_BE_NEEDED,
          MessageId: EVENT_METADATA_WHICH_SHOULD_NOT_BE_NEEDED,
          Message: message,
          MessageAttributes: EVENT_METADATA_WHICH_SHOULD_NOT_BE_NEEDED,
          Type: EVENT_METADATA_WHICH_SHOULD_NOT_BE_NEEDED,
          UnsubscribeUrl: EVENT_METADATA_WHICH_SHOULD_NOT_BE_NEEDED,
          TopicArn: EVENT_METADATA_WHICH_SHOULD_NOT_BE_NEEDED,
          Subject: EVENT_METADATA_WHICH_SHOULD_NOT_BE_NEEDED,
        },
      };
    }),
  };
};
