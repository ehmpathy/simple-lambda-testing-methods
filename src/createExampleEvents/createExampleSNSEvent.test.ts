import { SNSEvent } from 'aws-lambda';

import { createExampleSNSEvent } from './createExampleSNSEvent';

describe('createExampleSNSEvent', () => {
  it('should create an accurate looking sns event', () => {
    const event: SNSEvent = createExampleSNSEvent({
      messages: ['hello', 'there'],
    });
    expect(event).toMatchSnapshot();
  });
});
