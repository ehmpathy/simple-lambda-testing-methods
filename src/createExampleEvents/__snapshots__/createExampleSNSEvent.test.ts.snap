// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`createExampleSNSEvent should create an accurate looking sns event 1`] = `
{
  "Records": [
    {
      "EventSource": undefined,
      "EventSubscriptionArn": undefined,
      "EventVersion": undefined,
      "Sns": {
        "Message": "hello",
        "MessageAttributes": undefined,
        "MessageId": undefined,
        "Signature": undefined,
        "SignatureVersion": undefined,
        "SigningCertUrl": undefined,
        "Subject": undefined,
        "Timestamp": undefined,
        "TopicArn": undefined,
        "Type": undefined,
        "UnsubscribeUrl": undefined,
      },
    },
    {
      "EventSource": undefined,
      "EventSubscriptionArn": undefined,
      "EventVersion": undefined,
      "Sns": {
        "Message": "there",
        "MessageAttributes": undefined,
        "MessageId": undefined,
        "Signature": undefined,
        "SignatureVersion": undefined,
        "SigningCertUrl": undefined,
        "Subject": undefined,
        "Timestamp": undefined,
        "TopicArn": undefined,
        "Type": undefined,
        "UnsubscribeUrl": undefined,
      },
    },
  ],
}
`;
